app/components/GenericPlatform.tsx

```diff
@@ -198,16 +198,23 @@ export const GenericPlatform = ({
 
       const verifiedCredentials =
         selectedProviders.length > 0
-          ? verifyCredentialsResponse.credentials?.filter((cred: any) => !cred.error) || []
+          ? verifyCredentialsResponse.credentials?.filter((cred: any): cred is ValidResponseBody => !cred.error) || []
           : [];
 
       setVerificationResponse(verifyCredentialsResponse.credentials || []);
 
-      const stampPatches: StampPatch[] = platformProviderIds.map((provider: PROVIDER_ID) => {
-        const cred = verifiedCredentials.find((cred: any) => cred.record?.type === provider);
-        if (cred) return { provider, credential: cred.credential as VerifiableCredential };
-        else return { provider };
-      });
+      // If the stamp was selected and can be claimed, return the {provider, credential} to add the stamp
+      // If the stamp was not selected, return {provider} to delete the stamp
+      // If the stamp was selected but cannot be claimed, return null to do nothing and
+      //   therefore keep any existing valid stamp if it exists
+      const stampPatches: StampPatch[] = platformProviderIds
+        .map((provider: PROVIDER_ID) => {
+          const cred = verifiedCredentials.find((cred: any) => cred.record?.type === provider);
+          if (cred) return { provider, credential: cred.credential };
+          else if (!selectedProviders.includes(provider)) return { provider };
+          else return null;
+        })
+        .filter((patch): patch is StampPatch => Boolean(patch));
 
       await handlePatchStamps(stampPatches);
```

Comment:
I would add another test to spy and verify the payload of this function: `handlePatchStamps`